### PR TITLE
Adds optional support for new Account group association

### PR DESCRIPTION
**Pull request by bdemers**
If the property 'okta.application.groupId' is set, newly created accounts will be added to that group.
Otherwise the new account will be associated directly with the application.

### CODE REVIEW

File: extensions/spring/stormpath-spring/src/main/java/com/stormpath/spring/config/AbstractStormpathConfiguration.java [hunk 1]
```diff
@@ -292,6 +295,7 @@ public Application stormpathApplication() {
             oktaAppConfigMap.put("userApiQueryTemplate", apiSecretQueryTemplate);
             oktaAppConfigMap.put("applicationId", oktaApplicationId);
             oktaAppConfigMap.put("passwordPolicyName", oktaPasswordPolicyName);
+            oktaAppConfigMap.put("oktaApplicationGroupId", oktaApplicationGroupId);
```
Comment (dogeared) [inline extensions/spring/stormpath-spring/src/main/java/com/stormpath/spring/config/AbstractStormpathConfiguration.java:298]: Shouldn't all these use:  `OktaApplication.`, like `OktaApplication.APPLICATION_GROUP_ID`

File: impl/src/main/java/com/stormpath/sdk/impl/directory/OktaDirectory.java [hunk 2]
```diff
@@ -170,6 +171,24 @@ public Account createAccount(CreateAccountRequest request) {
...
+        return result;
+    }
+
+    protected void associateUser(Account account) {
+
+        String groupId = getStringProperty(OktaApplication.APPLICATION_GROUP_ID);
+        if (Strings.hasText(groupId)) {
+            associateUserWithGroup(account, groupId);
+        }
+        else {
```
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/directory/OktaDirectory.java]: nit - move else to previous line

**Comment by dogeared** [comment]
Minor nits. Feel free to update or not

**Review by dogeared — APPROVED** [review]

File: extensions/spring/stormpath-spring/src/main/java/com/stormpath/spring/config/AbstractStormpathConfiguration.java (hunk 1 above)
Comment (bdemers) [reply extensions/spring/stormpath-spring/src/main/java/com/stormpath/spring/config/AbstractStormpathConfiguration.java:298]: Nope, there is no dependency on `impl`, just a runtime dep

File: impl/src/main/java/com/stormpath/sdk/impl/directory/OktaDirectory.java (hunk 2 above)
Comment (bdemers) [reply impl/src/main/java/com/stormpath/sdk/impl/directory/OktaDirectory.java]: them is fighting words

**Pull request merged**